nip19.ts

```diff
@@ -48,3 +56,72 @@ function toHex(bech: string) {
     const data = new Uint8Array(bech32.fromWords(code.words));
     return utils.bytesToHex(data);
 }
+function encodeTLV(tlv: TLV): Uint8Array {
+    let entries: Uint8Array[] = [];
+
+    Object.entries(tlv).forEach(([t, vs]) => {
+        vs.forEach((v) => {
+            let entry = new Uint8Array(v.length + 2);
+            entry.set([parseInt(t)], 0);
+            entry.set([v.length], 1);
+            entry.set(v, 2);
+            entries.push(entry);
+        });
+    });
+
+    return utils.concatBytes(...entries);
+}
+function bytetoBech32(prefix: string, data: Uint8Array) {
+    const words = bech32.toWords(data);
+    return bech32.encode(prefix, words, 1500);
+}
+function parseTLV(data: Uint8Array): TLV {
+    let result: TLV = {};
+    let rest = data;
+    while (rest.length > 0) {
+        let t = rest[0];
+        let l = rest[1];
+        if (!l) throw new Error(`malformed TLV ${t}`);
+        let v = rest.slice(2, 2 + l);
+        rest = rest.slice(2 + l);
+        if (v.length < l) throw new Error(`not enough data to read on TLV ${t}`);
+        result[t] = result[t] || [];
+        result[t].push(v);
+    }
+    return result;
+}
+
+export class NaddrID {
```

Comment:
1. renane to `NAddr`
2. add `https://github.com/nostr-protocol/nips/blob/master/19.md#shareable-identifiers-with-extra-metadata` to comment. This is a good habit to include docs in the comment